src/RemoteResourceS3DecryptController.js

```diff
@@ -57,14 +57,24 @@ module.exports = class RemoteResourceS3DecryptController extends RemoteResourceS
   }
 
   async download(reqOpt) {
-
+    // disable response encoding, so that res.body is Buffer type
+    reqOpt.encoding = null;
+    // get full response payload to avoid http chunked data
+    reqOpt.resolveWithFullResponse = true;
     let res = await super.download(reqOpt);
     if (res.statusCode != 200) {
       return res;
     }
 
     let source = reqOpt.uri || reqOpt.url;
 
+    let isBinary = false;
+    if (res.headers['content-type'] === 'binary/octet-stream') {
```

Comment:
Have you confirmed we get binary/octet-stream for binay files?
What do we get for ascii armored files?

Reply:
for ascii, we get `application/octet-stream`. for binary we will get `binary/octet-stream`